cmake/scripts/buildinfo.cmake

```diff
@@ -19,19 +19,32 @@ if (NOT ETH_BUILD_PLATFORM)
 	set(ETH_BUILD_PLATFORM "unknown")
 endif()
 
-# Logic here: If prerelease.txt exists but is empty, it is a non-pre release.
-# If it does not exist, create our own prerelease string
+
 if (EXISTS ${ETH_SOURCE_DIR}/prerelease.txt)
-	file(READ ${ETH_SOURCE_DIR}/prerelease.txt SOL_VERSION_PRERELEASE)
-	string(STRIP "${SOL_VERSION_PRERELEASE}" SOL_VERSION_PRERELEASE)
+	message(FATAL_ERROR "The usage of the file prerelease.txt is deprecated. Please use the file cmake/prerelease.lock instead.")
+endif()
+
+if (SOLC_FORCE_RELEASE)
+	set(SOL_VERSION_PRERELEASE, "")
 else()
-	string(TIMESTAMP SOL_VERSION_PRERELEASE "develop.%Y.%m.%d" UTC)
-	string(REPLACE .0 . SOL_VERSION_PRERELEASE "${SOL_VERSION_PRERELEASE}")
+	if (SOLC_PRERELEASE_STRING)
+		set(SOL_VERSION_PRERELEASE "${SOLC_PRERELEASE_STRING}")
+	else()
+		if (EXISTS ${ETH_SOURCE_DIR}/cmake/prerelease.lock)
+			string(TIMESTAMP SOL_VERSION_PRERELEASE "develop.%Y.%m.%d" UTC)
+			string(REPLACE .0 . SOL_VERSION_PRERELEASE "${SOL_VERSION_PRERELEASE}")
+		else ()
+			set(SOL_VERSION_PRERELEASE, "")
+		endif()
+	endif()
 endif()
 
 if (EXISTS ${ETH_SOURCE_DIR}/commit_hash.txt)
-	file(READ ${ETH_SOURCE_DIR}/commit_hash.txt SOL_COMMIT_HASH)
-	string(STRIP ${SOL_COMMIT_HASH} SOL_COMMIT_HASH)
+	message(FATAL_ERROR "The usage of the file commit_hash.txt is deprecated. Please use the cmake option SOLC_COMMIT_HASH instead.")
+endif()
+
+if (SOLC_COMMIT_HASH)
+	set (SOL_COMMIT_HASH "${SOLC_COMMIT_HASH}")
```

Comment:
I just noticed that we now have some variables in `SOL_` and `SOLC_` variants.

Let's use `SOL_` for everything to avoid that. I updated the spec in the issue: https://github.com/ethereum/solidity/issues/9720#issuecomment-1059322965